lib/parser/source/comment/associator.rb

```diff
@@ -76,78 +76,111 @@ def initialize(ast, comments)
       #         [#<Parser::Source::Comment (string):3:8 "# bar">]
       #     }
       #
+      # NB: #associate is broken for nodes which have the same content.
+      # e.g. 2 identical lines in the code will see their comments "merged".
+      # Using map_using_locations=true prevents this. The returned hash uses
+      # node.location as a key to retrieve the comments for this node.
+      #
       # @return [Hash(Parser::AST::Node, Array(Parser::Source::Comment))]
+      #      or [Hash(Parser::Source::Map, Array(Parser::Source::Comment))]
       #
-      def associate
+      def associate(map_using_locations = false)
```

Comment:
This is not really what I meant... methods returning different types depending on an argument are horribly designed API.

Sigh. I'll just fix it myself